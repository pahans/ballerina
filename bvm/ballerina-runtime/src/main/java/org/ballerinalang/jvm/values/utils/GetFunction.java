/*
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.ballerinalang.jvm.values.utils;

import org.ballerinalang.jvm.values.ArrayValue;

/**
 * Functional interface for retrieving a value from an array. This is intended to abstract the different use cases of
 * ArrayValue class (i.e., as an array, as a tuple).
 *
 * @since 1.0
 */
public interface GetFunction {
    Object get(ArrayValue arr, long index);
}
