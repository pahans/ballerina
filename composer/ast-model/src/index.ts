export * from './ast-util';
export * from './model';
