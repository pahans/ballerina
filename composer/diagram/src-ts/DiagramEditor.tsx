import React from "react";

export interface DiagramEditorProps {
    width?: number;
}

interface DiagramEditorState {

}

export class DiagramEditor extends React.Component<DiagramEditorProps, DiagramEditorState> {
    
    render() {
        return <div>{'Diagram Editor'}</div>
    }
}