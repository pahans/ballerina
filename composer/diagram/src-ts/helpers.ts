import { createElement } from "react";
import ReactDOM from "react-dom";
import { Diagram, DiagramProps } from "./diagram";

export interface StaticDiagramArgs {
    target: HTMLElement;
    diagramProps: DiagramProps;
}

export const renderStaticDiagram = (args: StaticDiagramArgs) => {
    const BalDiagramElement = createElement(Diagram, args.diagramProps);
    ReactDOM.render(BalDiagramElement, args.target);
};
