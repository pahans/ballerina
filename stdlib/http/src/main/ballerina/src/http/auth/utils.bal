// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
//
// WSO2 Inc. licenses this file to you under the Apache License,
// Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import ballerina/log;
import ballerina/reflect;

# Auth annotation module.
const string ANN_MODULE = "ballerina/http";
# Resource level annotation name.
const string RESOURCE_ANN_NAME = "ResourceConfig";
# Service level annotation name.
const string SERVICE_ANN_NAME = "ServiceConfig";

# Represents the Authorization header name.
public const string AUTH_HEADER = "Authorization";

# Specifies how to send the authentication credentials when exchanging tokens.
public type CredentialBearer AUTH_HEADER_BEARER|POST_BODY_BEARER|NO_BEARER;

# Indicates that the authentication credentials should be sent via the Authentication header.
public const AUTH_HEADER_BEARER = "AUTH_HEADER_BEARER";

# Indicates that the Authentication credentials should be sent via the body of the POST request.
public const POST_BODY_BEARER = "POST_BODY_BEARER";

# Indicates that the authentication credentials should not be sent.
public const NO_BEARER = "NO_BEARER";

# Indicates the status code.
public const STATUS_CODE = "STATUS_CODE";

# Extracts the Authorization header value from the request.
#
# + req - Request instance
# + return - Value of the Authorization header
public function extractAuthorizationHeaderValue(Request req) returns @tainted string {
    // extract authorization header
    return req.getHeader(AUTH_HEADER);
}

# Tries to retrieve the inbound authentication handlers based on their hierarchy
# (i.e., first from the resource level and then from the service level, if it is not there at the resource level).
#
# + context - The `FilterContext` instance
# + return - Returns the authentication handlers or whether it is needed to engage listener-level handlers or not
function getAuthHandlers(FilterContext context) returns InboundAuthHandler[]|InboundAuthHandler[][]|boolean {
    ServiceResourceAuth? resourceLevelAuthAnn;
    ServiceResourceAuth? serviceLevelAuthAnn;
    [resourceLevelAuthAnn, serviceLevelAuthAnn] = getServiceResourceAuthConfig(context);

     // check if authentication is enabled for resource and service
    boolean resourceSecured = isServiceResourceSecured(resourceLevelAuthAnn);
    boolean serviceSecured = isServiceResourceSecured(serviceLevelAuthAnn);

    // if resource is not secured, no need to check further
    if (!resourceSecured) {
        log:printWarn("Resource is not secured. `enabled: false`.");
        return false;
    }
    // Checks if Auth providers are given at the resource level.
    if (resourceLevelAuthAnn is ServiceResourceAuth) {
        var resourceAuthHandlers = resourceLevelAuthAnn?.authHandlers;
        if (!(resourceAuthHandlers is ())) {
            return resourceAuthHandlers;
        }
    }

    // if service is not secured, no need to check further
    if (!serviceSecured) {
        log:printWarn("Service is not secured. `enabled: false`.");
        return true;
    }
    // No Auth providers found at the resource level. Thus, try at the service level.
    if (serviceLevelAuthAnn is ServiceResourceAuth) {
        var serviceAuthHandlers = serviceLevelAuthAnn?.authHandlers;
        if (!(serviceAuthHandlers is ())) {
            return serviceAuthHandlers;
        }
    }
    return true;
}

# Tries to retrieve the authorization scopes hierarchically - first from the resource level and then
# from the service level, if it is not there in the resource level.
#
# + context - `FilterContext` instance
# + return - Authorization scopes or whether it is needed to engage listener level scopes or not
function getScopes(FilterContext context) returns string[]|string[][]|boolean {
    ServiceResourceAuth? resourceLevelAuthAnn;
    ServiceResourceAuth? serviceLevelAuthAnn;
    [resourceLevelAuthAnn, serviceLevelAuthAnn] = getServiceResourceAuthConfig(context);

    // check if authentication is enabled for resource and service
    boolean resourceSecured = isServiceResourceSecured(resourceLevelAuthAnn);
    boolean serviceSecured = isServiceResourceSecured(serviceLevelAuthAnn);

    // if resource is not secured, no need to check further
    if (!resourceSecured) {
        return false;
    }
    // check if auth providers are given at resource level
    if (resourceLevelAuthAnn is ServiceResourceAuth) {
        var resourceScopes = resourceLevelAuthAnn?.scopes;
        if (!(resourceScopes is ())) {
            return resourceScopes;
        }
    }

    // if service is not secured, no need to check further
    if (!serviceSecured) {
        return true;
    }
    // no auth providers found in resource level, try in service level
    if (serviceLevelAuthAnn is ServiceResourceAuth) {
        var serviceScopes = serviceLevelAuthAnn?.scopes;
        if (!(serviceScopes is ())) {
            return serviceScopes;
        }
    }
    return true;
}

# Retrieve the authentication annotation value for resource level and service level.
#
# + context - The `FilterContext` instance
# + return - Returns the resource-level and service-level authentication annotations
function getServiceResourceAuthConfig(FilterContext context) returns [ServiceResourceAuth?, ServiceResourceAuth?] {
    // get authn details from the resource level
    any annData = reflect:getResourceAnnotations(context.getService(), context.getResourceName(), RESOURCE_ANN_NAME,
                                                 ANN_MODULE);
    ServiceResourceAuth? resourceLevelAuthAnn = ();
    if !(annData is ()) {
        HttpResourceConfig resourceConfig = <HttpResourceConfig> annData;
        resourceLevelAuthAnn = resourceConfig?.auth;
    }

    annData = reflect:getServiceAnnotations(context.getService(), SERVICE_ANN_NAME, ANN_MODULE);
    ServiceResourceAuth? serviceLevelAuthAnn = ();
    if !(annData is ()) {
        HttpServiceConfig serviceConfig = <HttpServiceConfig> annData;
        serviceLevelAuthAnn = serviceConfig?.auth;
    }

    return [resourceLevelAuthAnn, serviceLevelAuthAnn];
}

# Check for the service or the resource is secured by evaluating the enabled flag configured by the user.
#
# + serviceResourceAuth - Service or resource auth annotation
# + return - Whether the service or resource secured or not
function isServiceResourceSecured(ServiceResourceAuth? serviceResourceAuth) returns boolean {
    boolean secured = true;
    if (serviceResourceAuth is ServiceResourceAuth) {
        secured = serviceResourceAuth.enabled;
    }
    return secured;
}

# Creates a map out of the headers of the HTTP response.
#
# + resp - The `Response` instance
# + return - Returns the map of the response headers
function createResponseHeaderMap(Response resp) returns @tainted map<anydata> {
    map<anydata> headerMap = { STATUS_CODE: resp.statusCode };
    string[] headerNames = resp.getHeaderNames();
    foreach string header in headerNames {
        string[] headerValues = resp.getHeaders(<@untainted> header);
        headerMap[header] = headerValues;
    }
    return headerMap;
}

# Logs, prepares, and returns the `AuthenticationError`.
#
# + message -The error message
# + err - The `error` instance
# + return - Returns the prepared `AuthenticationError` instance
function prepareAuthenticationError(string message, error? err = ()) returns AuthenticationError {
    log:printDebug(function () returns string { return message; });
    if (err is error) {
        AuthenticationError preparedError = error(AUTHN_FAILED, message = message, cause = err);
        return preparedError;
    }
    AuthenticationError preparedError = error(AUTHN_FAILED, message = message);
    return preparedError;
}

# Logs, prepares, and returns the `AuthorizationError`.
#
# + message -The error message
# + err - The `error` instance
# + return - Returns the prepared `AuthorizationError` instance
function prepareAuthorizationError(string message, error? err = ()) returns AuthorizationError {
    log:printDebug(function () returns string { return message; });
    if (err is error) {
        AuthorizationError preparedError = error(AUTHZ_FAILED, message = message, cause = err);
        return preparedError;
    }
    AuthorizationError preparedError = error(AUTHZ_FAILED, message = message);
    return preparedError;
}
