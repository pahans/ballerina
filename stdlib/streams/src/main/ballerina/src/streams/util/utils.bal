// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
//
// WSO2 Inc. licenses this file to you under the Apache License,
// Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import ballerina/time;

# Creates  `streams:StreamEvent` object array for a record `t` received by the stream denoted by the name `streamNme`.
# + t - A record received by the stream `streamName`.
# + streamName - Name of the stream to which the record `t` is received.
# + return - Returns an array of streams:StreamEvents|()
public function buildStreamEvent(anydata t, string streamName) returns StreamEvent?[] {
    EventType evntType = "CURRENT";
    var keyVals = <map<anydata>>map<anydata>.constructFrom(t.clone());
    StreamEvent event = new([streamName, keyVals], evntType, time:currentTime().time);
    StreamEvent?[] streamEvents = [event];
    return streamEvents;
}

# Creates a RESET event from a given event.
# + event - The event from which the reset event is created.
# + return - A stream event of type streams:RESET.
public function createResetStreamEvent(StreamEvent event) returns StreamEvent {
    StreamEvent resetStreamEvent = new(event.data, "RESET", event.timestamp);
    return resetStreamEvent;
}

# Get the stream event from any? field. This function can only be used only if we are sure that the `anyEvent` is a
# streams:StreamEvent.
# + anyEvent - The object from which, the stream event is extracted.
# + return - Returns the extracted streams:StreamEvent object.
public function getStreamEvent(any? anyEvent) returns StreamEvent {
    return <StreamEvent>anyEvent;
}

# Converts a given array of streams:StreamEvent objects to an array of `streams:SnapshottableStreamEvent`.
# + events - The events to be coverted to snapshotable events.
# + return - Returns the converted snapshotable events.
public function toSnapshottableEvents(StreamEvent[]|any[]? events) returns SnapshottableStreamEvent?[] {
    SnapshottableStreamEvent?[] evts = [];
    if (events is StreamEvent[]) {
        foreach StreamEvent e in events {
            evts[evts.length()] = toSnapshottableEvent(e);
        }
    } else if (events is any[]) {
        foreach var e in events {
            if (e is StreamEvent) {
                evts[evts.length()] = toSnapshottableEvent(e);
            }
        }
    }
    return evts;
}

# Converts a given array of snapshotable events to an array of `streams:StreamEvent` objects.
# + events - Snapshotable events to be converted to `streams:StreamEvents`.
# + return - Returns the converted `streams:StreamEvents` objects array.
public function toStreamEvents(SnapshottableStreamEvent[]|any[]? events) returns StreamEvent?[] {
    StreamEvent?[] evts = [];
    if (events is SnapshottableStreamEvent[]) {
        foreach SnapshottableStreamEvent e in events {
            evts[evts.length()] = toStreamEvent(e);
        }
    } else if (events is any[]) {
        foreach var e in events {
            if (e is SnapshottableStreamEvent) {
                evts[evts.length()] = toStreamEvent(e);
            }
        }
    }
    return evts;
}

# Convert a single `streams:StreamEvent` object to `streams:SnapshottableStreamEvent` object.
# + event - The `streams:StreamEvent` object to be converted to snapshotable event.
# + return - The converted `streams:SnapshottableStreamEvent` object.
public function toSnapshottableEvent(StreamEvent event) returns SnapshottableStreamEvent {
    // todo: add the eventId
    return {
        eventType: event.eventType,
        timestamp: event.timestamp,
        data: event.data,
        streamName: event.streamName
    };
}

# Convert a single `streams:SnapshottableStreamEvent` object to `streams:StreamEvent` object.
# + event - The `streams:SnapshottableStreamEvent` object to be converted to a stream event.
# + return - The converted `streams:StreamEvent` object.
public function toStreamEvent(SnapshottableStreamEvent event) returns StreamEvent {
    // todo: add the eventId
    StreamEvent se = new(event.data, event.eventType, event.timestamp);
    return se;
}
