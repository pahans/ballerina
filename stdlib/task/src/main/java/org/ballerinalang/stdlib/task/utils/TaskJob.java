/*
 *  Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 *  WSO2 Inc. licenses this file to you under the Apache License,
 *  Version 2.0 (the "License"); you may not use this file except
 *  in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package org.ballerinalang.stdlib.task.utils;

import org.ballerinalang.stdlib.task.objects.ServiceInformation;
import org.ballerinalang.stdlib.task.objects.Task;
import org.quartz.Job;
import org.quartz.JobDataMap;
import org.quartz.JobExecutionContext;

import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_OBJECT;

/**
 * Represents a Quartz job related to an appointment.
 */
public class TaskJob implements Job {

    public TaskJob() {
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void execute(JobExecutionContext jobExecutionContext) {
        JobDataMap jobDataMap = jobExecutionContext.getMergedJobDataMap();
        Task task = (Task) jobDataMap.get(TASK_OBJECT);
        for (ServiceInformation serviceInformation : task.getServicesMap().values()) {
            TaskExecutor.executeFunction(serviceInformation);
        }
    }
}
