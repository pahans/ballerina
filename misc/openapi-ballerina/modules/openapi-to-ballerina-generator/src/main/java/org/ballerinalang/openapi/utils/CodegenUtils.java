/*
 * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.ballerinalang.openapi.utils;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.Operation;
import io.swagger.v3.oas.models.PathItem;

import java.io.IOException;
import java.io.PrintWriter;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Map;

import javax.annotation.Nullable;

/**
 * Utilities used by ballerina openapi code generator.
 */
public class CodegenUtils {

    /**
     * Is {@code path} is a valid ballerina project directory.
     *
     * @param path path to suspecting dir
     * @return if {@code path} is a ballerina project directory or not
     */
    public static boolean isBallerinaProject(Path path) {
        boolean isProject = false;
        Path cachePath = path.resolve("Ballerina.toml");

        // .ballerina cache path should exist in ballerina project directory
        if (Files.exists(cachePath)) {
            isProject = true;
        }

        return isProject;
    }

    /**
     * Resolves path to write generated main source files.
     *
     * @param pkg  module
     * @param path output path without module name
     * @return path to write generated source files
     */
    public static Path getSourcePath(String pkg, String path) {
        return (pkg == null || pkg.isEmpty()) ?
                Paths.get(path).resolve("src") :
                Paths.get(path).resolve("src").resolve(Paths.get(pkg));
    }

    /**
     * Resolves path to write generated implementation source files.
     *
     * @param pkg     module
     * @param srcPath resolved path for main source files
     * @return path to write generated source files
     */
    public static Path getImplPath(String pkg, Path srcPath) {
        return (pkg == null || pkg.isEmpty()) ? srcPath : srcPath.getParent();
    }

    /**
     * Writes a file with content to specified {@code filePath}.
     *
     * @param filePath valid file path to write the content
     * @param content  content of the file
     * @throws IOException when a file operation fails
     */
    public static void writeFile(Path filePath, String content) throws IOException {
        PrintWriter writer = null;

        try {
            writer = new PrintWriter(filePath.toString(), "UTF-8");
            writer.print(content);
        } finally {
            if (writer != null) {
                writer.close();
            }
        }
    }

    /**
     * Removes underscores and hyphens for identifiers.
     *
     * @param identifier Path
     * @return Cleaned identifier.
     */
    public static String normalizeForBIdentifier(@Nullable String identifier) {
        if (identifier == null) {
            return null;
        }

        String resourceName = identifier;
        if (identifier.split("-").length > 0) {
            String[] spllitedIdentifier = identifier.split("-");
            resourceName = spllitedIdentifier[spllitedIdentifier.length - 1];
        }

        return resourceName.replaceAll(" ", "_").replaceAll("-", "")
                .replaceAll("/", "");
    }

    /**
     * Generate operation ID using pattern "resource[number]".
     *
     * @param openAPI open api definition for the openapi
     * @return {@link String}operation ID which match to the "resource[number]"
     */
    public static String generateOperationId(OpenAPI openAPI) {
        int prevNumber = 0;
        for (Map.Entry<String, PathItem> path : openAPI.getPaths().entrySet()) {
            for (Operation operation : path.getValue().readOperations()) {
                String operationId = operation.getOperationId();
                if (operationId != null && operationId.matches("resource\\d+")) {
                    String[] numbers = operationId.split("resource");
                    if (numbers.length > 0) {
                        int number = Integer.parseInt(numbers[numbers.length - 1]);
                        if (prevNumber < number) {
                            prevNumber = number;
                        }
                    }
                }
            }
        }
        return "resource" + (prevNumber + 1);
    }
}
