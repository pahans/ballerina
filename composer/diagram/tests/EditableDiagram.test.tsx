import { BallerinaAST } from "@ballerina/ast-model";
import * as React from "react";
import { create } from "react-test-renderer";
import bbeASTs from "../resources/bbe-asts.json";
import { Diagram } from "../src-ts";
import { MockLangClient } from "./utils";

const bbeASTsArray = bbeASTs as Array<{ bbe: string, ast: BallerinaAST, title: string }>;

bbeASTsArray.forEach((bbeAST) => {
    test(`Diagram for ${bbeAST.title} renders properly`, () => {
        const component = create(
            <Diagram
              docUri={bbeAST.bbe}
              height={1000}
              width={1000}
              langClient={new MockLangClient(bbeAST.ast)}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
