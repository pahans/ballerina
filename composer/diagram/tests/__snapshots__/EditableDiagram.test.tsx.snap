// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagram for abstract_objects.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for anonymous_functions.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for anonymous_functions_test.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for any_type.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for any_type_test.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for arrays.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for arrays_test.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for async.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for async_test.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;

exports[`Diagram for ballerina_to_swagger.bal renders properly 1`] = `
<div>
  Diagram Editor
</div>
`;
